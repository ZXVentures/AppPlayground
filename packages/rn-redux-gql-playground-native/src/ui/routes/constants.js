export const MODAL = 'MODAL'
export const DEFAULT = 'DEFAULT'

export const VERTICAL = 'vertical'
export const HORIZONTAL = 'horizontal'

export const HOME = 'Home'
export const LEFT = 'Left'
export const RIGHT = 'Right'

export const directionByPush = {
  [MODAL]: VERTICAL,
  [DEFAULT]: HORIZONTAL
}
