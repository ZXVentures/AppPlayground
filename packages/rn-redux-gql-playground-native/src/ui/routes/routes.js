/* @flow */

import React from 'react' // eslint-disable-line no-unused-vars

import Home from '../scenes/Home'
import { DEFAULT_PUSH } from './pushConstants'

export const HOME_KEY = 'home'
export const DEBUG_KEY = 'debug'
export const USER_PROFILE_KEY = 'userProfile'

const routes = {
  home: {
    key: HOME_KEY,
    title: 'Home',
    pushType: DEFAULT_PUSH
  }
}

export default {
  ...routes,
  asScenes: {
    [HOME_KEY]: <Home />
  }
}
