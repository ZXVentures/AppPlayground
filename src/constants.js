export const APP_NAME = 'PlaygroundApp'
export const APP_REDUX_KEY = `${APP_NAME}:redux`
// eslint-disable-next-line global-require
export const APP_VERSION = require('../package.json').version
