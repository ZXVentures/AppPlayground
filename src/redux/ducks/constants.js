// Routing
export const PUSH = '@appPlayground/routes/PUSH'
export const POP = '@appPlayground/routes/POP'

// Localization
export const SET_LOCALE = '@ratebeer/intl/SET_LOCALE'
export const LOCALE_SHOULD_UPDATE = '@ratebeer/intl/LOCALE_SHOULD_UPDATE'
export const LOCALE_DID_UPDATE = '@ratebeer/intl/LOCALE_DID_UPDATE'
