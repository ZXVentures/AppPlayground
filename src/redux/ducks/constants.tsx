

// Localization
export const SET_LOCALE = '@ratebeer/intl/SET_LOCALE'
export const LOCALE_SHOULD_UPDATE = '@ratebeer/intl/LOCALE_SHOULD_UPDATE'
export const LOCALE_DID_UPDATE = '@ratebeer/intl/LOCALE_DID_UPDATE'
