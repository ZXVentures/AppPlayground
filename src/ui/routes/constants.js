export const VERTICAL = 'vertical'
export const HORIZONTAL = 'horizontal'

export const HOME = 'Home'
export const LEFT = 'Left'
export const MODAL = 'Modal'
export const RIGHT = 'Right'

export const directionByPush = {
  [MODAL]: VERTICAL,
  [DEFAULT]: HORIZONTAL
}
