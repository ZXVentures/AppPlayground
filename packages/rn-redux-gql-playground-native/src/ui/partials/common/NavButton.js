import React, { React$Element, Component } from 'react' // eslint-disable-line no-unused-vars

import { TouchableHighlight, Text } from 'react-native'

import { LEFT, RIGHT } from '../../routes/constants'

export class NavButton extends Component {
  render(): React$Element {
    return (
      <TouchableHighlight style={{
        flex: 1,
        flexDirection: 'row',
        alignItems: 'center',
        padding: 10
      }}
        onPress={this._onPress.bind(this)}>
        <Text style={{
          fontSize: 16,
          fontWeight: '400'
        }}>{this.props.title}</Text>
      </TouchableHighlight>
    )
  }

  _onPress() {
    switch (this.props.title) {
    case LEFT: return this.props.showLeft()
    case RIGHT: return this.props.showRight()
    default: return this.props.pop()
    }
  }
}

import { showLeft, showRight, pop } from '../../../redux/ducks/nav'
import { connect } from 'react-redux'

const mapDispatchToProps = (dispatch: Function): Object => ({
  showLeft: (): void => dispatch(showLeft()),
  showRight: (): void => dispatch(showRight()),
  pop: (): void => dispatch(pop())
})

export default connect(null, mapDispatchToProps)(NavButton)
