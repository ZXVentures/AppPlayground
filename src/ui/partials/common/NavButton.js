import React, { React$Element, Component } from 'react' // eslint-disable-line no-unused-vars

import { TouchableHighlight } from 'react-native'

import { LEFT, RIGHT } from '../../routes/constants'

import Icon from 'react-native-vector-icons/FontAwesome'

export class NavButton extends Component {
  render(): React$Element {
    const iconName = (this.props.title === LEFT) ? 'bars' : 'globe'

    return (
      <TouchableHighlight style={{
        flex: 1,
        flexDirection: 'row',
        alignItems: 'center',
        padding: 10
      }}
        onPress={this._onPress.bind(this)}>
        <Icon name={iconName}
          size={24}
          color={'#64b5f6'}
        />
      </TouchableHighlight>
    )
  }

  _onPress() {
    switch (this.props.title) {
    case LEFT: return this.props.showLeft()
    case RIGHT: return this.props.showRight()
    default: return this.props.pop()
    }
  }
}

import { showLeft, showRight, pop } from '../../../redux/ducks/nav'
import { connect } from 'react-redux'

const mapDispatchToProps = (dispatch: Function): Object => ({
  showLeft: (): void => dispatch(showLeft()),
  showRight: (): void => dispatch(showRight()),
  pop: (): void => dispatch(pop())
})

export default connect(null, mapDispatchToProps)(NavButton)
