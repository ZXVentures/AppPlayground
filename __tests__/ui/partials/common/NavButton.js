import 'react-native'
import React from 'react' // eslint-disable-line no-unused-vars

import { NavButton, mapDispatchToProps } from 'ui/partials/common/NavButton'

import { LEFT, RIGHT } from 'ui/routes/constants'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

describe('ui/partials/common/NavButton', () => {
  it('snapshot test', () => {
    const tree = renderer.create(
      <NavButton />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('handles press actions appropriately', () => {
    const navButton = new NavButton()
    const mockFn = jest.fn()
    navButton.props = {}
    navButton.props.showLeft = mockFn
    navButton.props.showRight = mockFn
    navButton.props.pop = mockFn

    navButton.props.title = LEFT
    navButton._onPress()
    expect(mockFn).toHaveBeenCalled()

    mockFn.mockClear()

    navButton.props.title = RIGHT
    navButton._onPress()
    expect(mockFn).toHaveBeenCalled()

    mockFn.mockClear()

    navButton.props.title = 'TEST'
    navButton._onPress()
    expect(mockFn).toHaveBeenCalled()

    mockFn.mockClear()
  })

  it('maps dispatch to props', () => {
    const mockDispatch = jest.fn()

    const props = mapDispatchToProps(mockDispatch)

    expect(props).not.toBeNull()
    expect(Object.keys(props).length).toBe(3) // you need to add a test if this fails
    props.showLeft()
    expect(mockDispatch).toHaveBeenCalled()

    mockDispatch.mockClear()

    props.showRight()
    expect(mockDispatch).toHaveBeenCalled()

    mockDispatch.mockClear()

    props.pop()
    expect(mockDispatch).toHaveBeenCalled()

    mockDispatch.mockClear()
  })
})
