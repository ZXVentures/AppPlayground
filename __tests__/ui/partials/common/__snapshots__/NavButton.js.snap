exports[`ui/partials/common/NavButton snapshot test 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "padding": 10,
      },
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#64b5f6",
          "fontSize": 24,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }>
    ï‚¬
  </Text>
</View>
`;
